play-akka-http-ws-standalone/src/main/scala/play/api/libs/ws/akkahttp/StandaloneAkkaHttpWSRequest.scala

```diff
@@ -0,0 +1,302 @@
+
+  /**
+   * The virtual host this request will use
+   */
+  override def virtualHost: Option[String] = ???
+
+  /**
+   * The proxy server this request will use
+   */
+  override def proxyServer: Option[WSProxyServer] = ???
+
+  /**
+   * sets the signature calculator for the request
+   *
+   * @param calc the signature calculator
+   */
+  override def sign(calc: WSSignatureCalculator): Self = ???
+
+  /**
+   * sets the authentication realm
+   */
+  override def withAuth(username: String, password: String, scheme: WSAuthScheme): Self =
+    scheme match {
+      case WSAuthScheme.BASIC => copy(request = request.addHeader(Authorization(BasicHttpCredentials(username, password))))
+      case authScheme => throw new IllegalArgumentException(s"Authentication scheme [$scheme] not yet supported")
+    }
+
+  /**
+   * Returns this request with the given headers, discarding the existing ones.
+   *
+   * @param headers the headers to be used
+   */
+  override def withHttpHeaders(headers: (String, String)*): Self =
+    copy(request = request.withHeaders(headers
+      .map { case (name, value) => HttpHeader.parse(name, value) }
+      .collect {
+        case ParsingResult.Ok(header, _) => header
+      }
+      .to[immutable.Seq]
+    ))
+
+  /**
+   * Returns this request with the given query string parameters, discarding the existing ones.
+   *
+   * @param parameters the query string parameters
+   */
+  override def withQueryStringParameters(parameters: (String, String)*): Self =
+    copy(request = request.copy(uri = request.uri.withQuery(Uri.Query(parameters.reverse: _*))))
+
+  /**
+   * Returns this request with the given cookies, discarding the existing ones.
+   *
+   * @param cookies the cookies to be used
+   */
+  override def withCookies(cookies: WSCookie*): Self =
+    copy(request = request.mapHeaders(h => h.filter {
+      case c: Cookie => false
+      case _ => true
+    } ++ cookies.map(c => Cookie(c.name, c.value))))
+
+  /**
+   * Sets whether redirects (301, 302) should be followed automatically
+   */
+  override def withFollowRedirects(follow: Boolean): Self = ???
+
+  /**
+   * Sets the maximum time you expect the request to take.
+   * Use Duration.Inf to set an infinite request timeout.
+   * Warning: a stream consumption will be interrupted when this time is reached unless Duration.Inf is set.
+   */
+  override def withRequestTimeout(timeout: Duration): Self =
+    copy(timeout = timeout)
+
+  /**
+   * Adds a filter to the request that can transform the request for subsequent filters.
+   */
+  override def withRequestFilter(filter: WSRequestFilter): Self =
+    copy(filters = filters :+ filter)
+
+  /**
+   * Sets the virtual host to use in this request
+   */
+  override def withVirtualHost(vh: String): Self =
+    copy(request = request.addHeader(Host(vh)))
+
+  /**
+   * Sets the proxy server to use in this request
+   */
+  override def withProxyServer(proxyServer: WSProxyServer): Self = ???
```

Comment:
missing feature in our impl right? AFAIR we still only do https proxy but not any http?

Reply:
Add fixme and link to ticket in akka-http